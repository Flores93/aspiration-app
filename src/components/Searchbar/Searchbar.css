.search-bar {
  width: 20vw;
  height: 30px;
  border: 0px;
  background-color: rgb(219, 219, 219);
  position: relative;
}
.search-button {
  position: inherit;
  cursor: pointer;
  height: 30px;
  border: 0px;
  color: rgb(104, 104, 104);
  font-weight: bold;
  background-color: rgb(223, 230, 255);
}

.search-button:hover {
  background-color: rgb(70, 70, 201);
  color: white;
  transition: 0.3s;
}
