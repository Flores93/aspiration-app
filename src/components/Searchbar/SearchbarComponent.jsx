import { useState } from "react";

import "./Searchbar.css";
const Searchbar = ({ onSearch, initSearchValue }) => {
  const [inputValue, setInputValue] = useState(initSearchValue);
  return (
    <nav>
      <form onSubmit={(e) => onSearch(e, inputValue, setInputValue)}>
        <input
          className="search-bar"
          type="search"
          value={inputValue}
          onChange={(e) => setInputValue(e.target.value)}
        />
        <button className="search-button" type="submit">
          Search
        </button>
      </form>
    </nav>
  );
};

export default Searchbar;
