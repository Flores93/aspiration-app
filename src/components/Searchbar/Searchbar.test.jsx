import { mount } from "enzyme";

import Searchbar from "./SearchbarComponent";

describe("Searchbar Component Test", () => {
  const onSearchMock = jest.fn();

  const wrapper = mount(
    <Searchbar onSearch={onSearchMock} initSearchValue="React" />
  );

  test("should show the component correctly", () => {
    const searchbarExists = wrapper.find(".search-bar").exists();
    const buttonExists = wrapper.find(".search-button").exists();

    expect(wrapper).toMatchSnapshot();
    expect(searchbarExists).toBe(true);
    expect(buttonExists).toBe(true);
  });

  test("should call the onSearch function when form is submitted", () => {
    wrapper.find("form").prop("onSubmit")({
      preventDefault() {},
    });

    expect(onSearchMock).toHaveBeenCalled();
  });
});
