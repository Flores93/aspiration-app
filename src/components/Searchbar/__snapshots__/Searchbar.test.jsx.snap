// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Component Test should show the component correctly 1`] = `
<nav>
  <form
    onSubmit={[Function]}
  >
    <input
      className="search-bar"
      onChange={[Function]}
      type="search"
      value="React"
    />
    <button
      className="search-button"
      type="submit"
    >
      Search
    </button>
  </form>
</nav>
`;
